packages/migration-graph-shared/test/entities/project-graph.test.ts

```diff
@@ -96,6 +96,94 @@ describe('project-graph', () => {
       expect(barNode.adjacent.has(bazNode)).toBe(true);
       expect(barNode.adjacent.has(blorpNode)).toBe(true);
     });
+    test('should not include a file out of the package scope', () => {
+      const files = {
+        packages: {
+          branch: {
+            'package.json': `{
+              "name": "@some-workspace/branch",
+              "version": "1.0.0",
+              "main": "index.js",
+              "dependencies": {
+                "@some-workspace/leaf": "*"
+              }
+            }`,
+            'index.js': `
+              import { do } from '@some-workspace/leaf';
+              import './lib/a';
+            `,
+            'build.js': `import '../../some-shared-util';`,
+            lib: {
+              'a.js': `
+              // a.js
+              console.log('foo');
+             `,
+            },
+          },
+          leaf: {
+            'package.json': `{
+              "name": "@some-workspace/leaf",
+              "version": "1.0.0",
+              "main": "index.js"
+            }`,
+            'index.js': `
+              import './lib/impl';
+              export function do() { console.log(''); }
+            `,
+            'build.js': `import '../../some-shared-util';`,
+            lib: {
+              'impl.js': `
+                // impl.js
+              `,
+            },
+          },
+        },
+        'some-shared-util.js': '// something-shared',
+        'package.json': `
+          {
+            "name": "root-package",
+            "version": "1.0.0",
+            "main": "index.js",
+            "license": "MIT",
+            "workspaces": [
+              "packages/*"
+            ]
+          }    
+        `,
+      };
+
+      // mutate a file to include a file from the root directory.
+
+      const baseDir = create(files);
+
+      const projectGraph = new ProjectGraph(baseDir);
+
+      projectGraph.discover();
+
+      const rootNode = projectGraph.graph.getNode('root-package');
+      const branchNode = projectGraph.graph.getNode('@some-workspace/branch');
+      const leafNode = projectGraph.graph.getNode('@some-workspace/leaf');
+
+      // Validate edges
+      expect(rootNode.adjacent.has(branchNode)).toBe(true);
+      expect(rootNode.adjacent.has(leafNode)).toBe(true);
+      expect(branchNode.adjacent.has(leafNode)).toBe(true);
+
+      // Validate graph order correctness
+      const nodes = projectGraph.graph.topSort().map((node) => node.content.pkg);
+
+      expect(nodes.length).toBe(3);
+
+      expect(nodes[0].packageName).toBe('@some-workspace/leaf');
+      expect(flatten(nodes[0].getModuleGraph().topSort())).toStrictEqual([
+        'lib/impl.js',
+        'index.js',
```

Comment:
This should be a TODO or not included because the expected output doesn't include the `build.js` file because we currently scope all packages to traverse `index.js` only, rather than `.`. This is the test case for a workspace with a out of bounds file.



In a future PR i will be making the default entrypoint for a package `.` which will help in the use case.